### PR TITLE
[refactor] #1725: Remove global state in Torii

### LINKED ISSUE #1725: Remove Torii global state
As it is implemented at the moment, Torii has global state shared by all filters due to the limitations of the `add_state` method. The code is somewhat convoluted because there is the same set of variables in both `Torii` and `ToriiState`. 

Make changes to `add_state`(implement it as `macro_rules`) to support fine tuning of what filters receive as state

### PR DESCRIPTION
**Pull request by mversic**
### Description of the Change

Have `warp` filters in `Torii` only receive part of the state they require, not everything

### Issue

Closes  #1725

### CODE REVIEW

**Review by s8sato — DISMISSED** [review]

File: core/src/torii/utils.rs [hunk 1]
```diff
@@ -25,13 +25,13 @@ impl<T: Encode + Send> Reply for Scale<T> {
...
 
 /// Adds state to filter
-pub fn add_state<State: Send + Clone>(
-    state: State,
-) -> impl Filter<Extract = (State,), Error = Rejection> + Clone + Send {
-    warp::any().and_then(move || {
-        let state = state.clone();
-        async move { Ok::<_, Rejection>(state) }
-    })
+macro_rules! add_state {
```
Comment (appetrosyan) [inline core/src/torii/utils.rs:28]: Any particular reason to use a declarative macro instead of a templated function?

Comment (mversic) [reply core/src/torii/utils.rs:28]: I don't think it's possible to create variadic functions in rust

Comment (appetrosyan) [reply core/src/torii/utils.rs:28]: https://github.com/rust-lang/rust/issues/44930

yeah. It's not been finished yet (much less stabilised).

**Review by appetrosyan — APPROVED** [review]

File: core/src/torii/mod.rs [hunk 2]
```diff
@@ -240,92 +223,80 @@ impl<W: WorldTrait> Torii<W> {
...
-            Ok(())
-        }
-        Err(e) => {
-            iroha_logger::error!("Status address configuration parse error");
-            Err(eyre::Error::new(e))
+    /// Start status endpoint.
+    ///
+    /// # Errors
+    /// Can fail due to listening to network or if http server fails
+    async fn start_status(self) -> eyre::Result<()> {
```
Comment (eadventurous) [inline core/src/torii/mod.rs:259]: We only need `self.wsv` and `self.network` here, maybe it is best not to move `self` into func? Just take them as arguments.

I think there is a high probability that we might need to start services on other ports.

File: core/src/torii/utils.rs (hunk 1 above)
Comment (eadventurous) [reply core/src/torii/utils.rs:28]: Yeah, macros are pretty much the standard way to handle these cases in Rust at the moment.

File: core/src/torii/mod.rs (hunk 2 above)
Comment (mversic) [reply core/src/torii/mod.rs:259]: mhm, I thought this was like a chain of calls to start services. One service is started and then forwards `self` to another to start itself and so on

Comment (eadventurous) [reply core/src/torii/mod.rs:259]: But as the task is spawned, basically we move torii inside this spawned task and consume it, the way it is currently implemented.

Comment (s8sato) [reply core/src/torii/mod.rs:259]: When it comes to further services on different ports in the future,
I prefer existing(flat) spawns to chained(nested) spawns for readability and modularity

Comment (mversic) [reply core/src/torii/mod.rs:259]: I'm going to use flat spawns, will commit changes soon

**Review by appetrosyan — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**